web/src/routes/api/account/gmail/remove-email-labels/+server.ts

```diff
@@ -0,0 +1,198 @@
+}
+
+/**
+ * Helper function that retrieves the label IDs of a Gmail thread (message).
+ *
+ * @async
+ * @param {string} accessToken - A valid access token for the Gmail API.
+ * @param {string} threadId - The ID of the Gmail thread to retrieve label IDs for.
+ * @returns {Promise<string[]>} - A promise that resolves with an array of label IDs for the given thread.
+ * @throws Throws an error if the API request fails or if the response is invalid.
+ */
+const getThreadLabels = async (accessToken: string, threadId: string): Promise<string[]> => {
+
+  // The Gmail thread endpoint, gives the details about a specific Gmail thread
+  const endpoint = `https://gmail.googleapis.com/gmail/v1/users/me/threads/${threadId}`;
+  
+  // query the thread endpoint
+  const response = await fetch(endpoint, {
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      Accept: 'application/json',
+    },
+  });
+
+  // Parse the API response data
+  const data = await response.json();
+  
+  // if the thread has messages with labels, return them
+  if (data.messages) {
+    return data.messages[0].labelIds
+  } else {
+    return [];
+  }
+}
+
+/**
+ * Remove labels from a Gmail thread using the Gmail API.
+ * 
+ * @param {string} accessToken - The access token for the authenticated user.
+ * @param {string} threadId - The ID of the Gmail thread to modify.
+ * @param {string[]} labelIdsToRemove - An array of label IDs to remove from the thread.
+ * @returns {Promise<boolean>} A boolean indicating whether the labels were successfully removed from the thread.
+ */
+const removeLabelsFromThread = async (accessToken: string, threadId: string, labelIdsToRemove: string[]): Promise<boolean> => {
+    
+  // The Gmail modify thread endpoint which will allow us to remove labels from the given thread
+  const endpoint = `https://gmail.googleapis.com/gmail/v1/users/me/threads/${threadId}/modify`;
+
+  // Get the labels before the update
+  const labelsBeforeUpdate = await getThreadLabels(accessToken, threadId);
+
+  // POST to the modify endpoint with the array of labels to remove
+  const response = await fetch(endpoint, {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        removeLabelIds: labelIdsToRemove,
+        addLabelIds: [],
+    }),
+  });
+
+  // Get the labels after the update
+  const labelsAfterUpdate = await getThreadLabels(accessToken, threadId);
+
+  // Calculate what is expected of the updated labels
+  const expectedLabels = labelsBeforeUpdate.filter(label => !labelIdsToRemove.includes(label));
+
+  // Check if the labels retrieved via the Gmail API match what is expected
+  const updateSuccessful = (
+    expectedLabels.length === labelsAfterUpdate.length &&
+    expectedLabels.every((value, index) => value === labelsAfterUpdate[index])
+  );
+
+  // Return a boolean indicating whether the update was successful
```

Comment:
I don't think this is necessary. Let's assume Google's API is trustworthy. If the response is `ok`, then we can assume the update was successful